akka-http-tests/src/test/scala/akka/http/server/directives/BasicDirectivesSpec.scala

```diff
@@ -26,4 +29,22 @@ class BasicDirectivesSpec extends RoutingSpec {
       } ~> check { responseAs[String] shouldEqual "GET" }
     }
   }
-}
\ No newline at end of file
+
+  "The `extractUri` directive" should {
+    "extract from the request url" in {
+      Get("https://example.com/foo") ~> {
+        extractUri { echoComplete }
+      } ~> check { responseAs[String] shouldEqual "https://example.com/foo" }
+    }
+    "extract from a X-Request-URI header" in {
+      Get() ~> addHeader(RawHeader("x-request-uri", "http://example.com/foo")) ~> {
+        extractUri { echoComplete }
+      } ~> check { responseAs[String] shouldEqual "http://example.com/" }
```

Comment:
Can you point me to how exactly the header value ends up in uri? I couldn't figure it out.